Add prometheus integration

stonepreston opened the pull request:
prometheus integration requires patching the juju-created service to expose the metrics port on 8888. A custom scrape job must be defined since the metrics port is different than the default port expected by the [prometheus_scrape library](https://charmhub.io/prometheus-k8s/libraries/prometheus_scrape). 

Added some documentation regarding the relevant relations that are now available as well.

nsklikas commented:
LGTM, can you fix the tests before we merge it? I think you just need to patch the ServicePatch class and rebase

Commit:
Add prometheus integration

stonepreston committed:
Merge branch 'sidecar-migration' into scp/prometheus-integration

Commit:
fix linting

Commit:
Fix failing unit tests

Commit:
Use reusable workflows

Commit:
Add wokeignore

stonepreston commented:
I added the shared workflows to the workflows file to bring this in line with the other CK repositories. Note that the ServicePatch lib requires python 3.8+, so I removed 3.6 and 3.7 from the test matrix.

Commit:
Remove v0 service patch lib
